yarn-project/ethereum/src/utils.ts

```diff
@@ -64,3 +66,110 @@ function tryExtractEvent<
+      return hex;
+    }
+    if (!hex.startsWith('0x')) {
+      return hex;
+    }
+    if (hex.length <= length * 2) {
+      return hex;
+    }
+    return `${hex.slice(0, length)}...${hex.slice(-length)}`;
+  };
+
+  const formatRequestBody = (body: string) => {
+    try {
+      const parsed = JSON.parse(body);
+
+      // Recursively process all parameters that might contain hex strings
+      const processParams = (obj: any): any => {
+        if (Array.isArray(obj)) {
+          return obj.map(item => processParams(item));
+        }
+        if (typeof obj === 'object' && obj !== null) {
+          const result: any = {};
+          for (const [key, value] of Object.entries(obj)) {
+            result[key] = processParams(value);
+          }
+          return result;
+        }
+        if (typeof obj === 'string') {
+          if (obj.startsWith('0x')) {
+            return truncateHex(obj);
+          }
+        }
+        return obj;
+      };
+
+      // Process the entire request body
+      const processed = processParams(parsed);
+      return JSON.stringify(processed, null, 2);
+    } catch {
+      return body;
+    }
+  };
+
+  const truncateHexStringsInText = (text: string): string => {
+    const hexRegex = /0x[a-fA-F0-9]{10,}/g;
+    return text.replace(hexRegex, hex => truncateHex(hex));
+  };
+
+  const extractAndFormatRequestBody = (message: string): string => {
+    // First handle Request body JSON
+    const requestBodyRegex = /Request body: ({[\s\S]*?})\n/g;
+    let result = message.replace(requestBodyRegex, (match, body) => {
+      return `Request body: ${formatRequestBody(body)}\n`;
+    });
+
+    // Then handle Arguments section
+    const argsRegex = /((?:Request |Estimate Gas )?Arguments:[\s\S]*?(?=\n\n|$))/g;
+    result = result.replace(argsRegex, section => {
+      const lines = section.split('\n');
+      const processedLines = lines.map(line => {
+        // Check if line contains a colon followed by content
+        const colonIndex = line.indexOf(':');
+        if (colonIndex !== -1) {
+          const [prefix, content] = [line.slice(0, colonIndex + 1), line.slice(colonIndex + 1).trim()];
+          // If content contains a hex string, truncate it
+          if (content.includes('0x')) {
+            const hexMatches = content.match(/0x[a-fA-F0-9]+/g) || [];
+            let processedContent = content;
+            hexMatches.forEach(hex => {
+              processedContent = processedContent.replace(hex, truncateHex(hex));
+            });
+            return `${prefix} ${processedContent}`;
+          }
+        }
+        return line;
+      });
+      return processedLines.join('\n');
+    });
+
+    // Finally, catch any remaining hex strings in the message
+    result = truncateHexStringsInText(result);
+
+    return result;
+  };
+
+  return JSON.stringify({ error: extractAndFormatRequestBody(error?.message || String(error)) }, null, 2).replace(
+    /\\n/g,
+    '\n',
+  );
```

Comment:
Should we check this error is actually a viem error before running all this logic? Also, let's add a big try/catch, so we don't throw again while trying to format the error if something goes wrong.